front/src/components/Detail.jsx

```diff
@@ -1,14 +1,12 @@
-import { Link, useLocation } from "react-router-dom";
+import { Link } from "react-router-dom";
 import React, { useState } from "react";
 import useQuestions from "components/Questions";
 
 const Detail = () => {
-  const location = useLocation();
-  const pathID = location.pathname.slice(-1);
-  const pageQuestion = useQuestions(pathID);
+  const slug = window.location.href.substring(window.location.href.lastIndexOf("/") + 1);
```

Comment:
Is this to just grab the PK of the question off the URL? 

1.) We don't like this because of magic numbers 
2.) We don't like this because it requires that the url never changes. This makes this code brittle and harder to maintain. Will you always be sure that the item between the `/<whatever>/` and the end of the url is what we want? What if an engineer in the future adds `/<whatever>/<other_thing>`. This is what it means for code to be maintainable this introduces a failure point.

We want to pass this as a prop to detail i.e. `const Detail = (slug) => {}` right up top. Lets use react to get this! If we absolutely _must_ use the url to get it lets use react router to make it a url object and get it from there, not just string splitting.

Reply:
Very interesting point! Agree, code independent of the URL will be better. I’ve spent quite some time researching props in react and there’s little information about how to use them with in functional components outside of the function return (which I believe I need to do for this particular logic). Although, in my research, I found another means of passing data/props between components and that's through entering a value in the state property of the react-router-dom location object. This capability exists for the methods fthat I use to change pages such as Link and useHistory.push. I can access the data in the next component via the useLocation hook. See upcoming commit.